### PR TITLE
Iterate on propagate

### PR DESCRIPTION
**Pull request by SimonBiggs**
It was breaking when merged into main

### CODE REVIEW

**Comment by Matthew-Jennings** [comment]
Hi @SimonBiggs, I see you've requested my review but the test is still failing. Are you still working on a solution?

File: .github/workflows/library.yml [hunk 1]
```diff
@@ -364,8 +364,8 @@ jobs:
           git config --local user.email "propagate-bot@pymedphys.com"
           git config --local user.name "PyMedPhys Propagate Bot"
 
-          git remote set-url origin "https://github.com/${REPO}.git"
-          git fetch origin $BRANCH
+          git remote set-url upstream "https://github.com/${REPO}.git"
```
Comment (SimonBiggs) [inline .github/workflows/library.yml:367]:
```suggestion
          git remote add upstream "https://github.com/${REPO}.git"
```

File: .github/workflows/library.yml [hunk 2]
```diff
@@ -364,8 +364,8 @@ jobs:
           git config --local user.email "propagate-bot@pymedphys.com"
           git config --local user.name "PyMedPhys Propagate Bot"
 
-          git remote set-url origin "https://github.com/${REPO}.git"
-          git fetch origin $BRANCH
+          git remote add upstream "https://github.com/${REPO}.git"
+          git fetch upstream $BRANCH || git fetch origin $BRANCH
```
Comment (SimonBiggs) [inline .github/workflows/library.yml:368]:
```suggestion
          (git fetch upstream $BRANCH && git checkout upstream/$BRANCH) || (git fetch origin $BRANCH && git checkout origin/$BRANCH)
```

File: .github/workflows/library.yml [hunk 3]
```diff
@@ -364,8 +364,8 @@ jobs:
...
           git checkout $BRANCH
```
Comment (SimonBiggs) [inline .github/workflows/library.yml:369]:
```suggestion
```

**Comment by SimonBiggs** [comment]
> I see you've requested my review but the test is still failing. Are you still working on a solution?

Yeah, sorry about that. I thought I had fixed it... to no avail. Hopefully this time :slightly_smiling_face:.

**Pull request merged**